src/components/PageSettingsModal/ResourcePageSettingsModal.tsx

```diff
@@ -172,16 +173,21 @@ export const ResourcePageSettingsModal = ({
         })
         setValue("permalink", "")
       }
-      if (
-        pageData.content.frontMatter.layout === "link" &&
-        pageData.content.frontMatter.permalink
-      ) {
-        // remove https:// from resource pages with external permalinks
-        setValue(
-          "permalink",
-          pageData.content.frontMatter.permalink.replace("https://", "")
-        )
+      if (pageData.content.frontMatter.layout === "link") {
+        if (pageData.content.frontMatter.permalink) {
+          // backwards compatible with previous link files
+          setValue(
+            "external",
+            pageData.content.frontMatter.permalink.replace("https://", "")
+          )
```

Comment:
after we `setValue`, should we do a `delete processedData.permalink` rather than during the `onSubmit`?

Reply:
Hmm `processedData` is only retrieved on submission! The alternative here would be to modify `pageData.content.frontMatter.permalink` directly, but that causes issues because the value is retrieved via react-query - I think it might be better to leave it on the `onSubmit`, since we'll still need to it handle legacy `file` type resources as well. What are your thoughts?